plugins/modules/file_interface.py

```diff
@@ -0,0 +1,465 @@
+                    ip_address=ip_address)
+                if len(file_interfaces) == 0:
+                    return None
+                else:
+                    file_interface_details = file_interfaces[0]
+
+            msg = f'Successfully got file_interface details {file_interface_details}'
+            LOG.info(msg)
+            return file_interface_details
+
+        except Exception as e:
+            msg = (f'Get file interface details for PowerStore array name '
+                   f': {self.cluster_name} , global id : '
+                   f'{self.cluster_global_id} failed with error {str(e)} ')
+            if isinstance(e, utils.PowerStoreException) and \
+                    e.err_code == utils.PowerStoreException.HTTP_ERR \
+                    and e.status_code == "404":
+                LOG.info(msg)
+                return None
+            LOG.error(msg)
+            self.module.fail_json(msg=msg, **utils.failure_codes(e))
+
+    def is_modify_required(self, file_interface_details, file_interface_params):
+        """To get the details of the fields to be modified."""
+
+        msg = f'File interface details: {file_interface_details}'
+        LOG.info(msg)
+        modify_dict = dict()
+
+        modify_keys = ["ip_address", "prefix_length", "gateway",
+                       "vlan_id", "is_disabled", "ip_port_id",
+                       "is_destination_override_enabled"]
+        for key in modify_keys:
+            if file_interface_params[key] is not None and \
+                    file_interface_params[key] != file_interface_details[key]:
+                modify_dict[key] = file_interface_params[key]
+        if modify_dict:
+            return modify_dict
+        else:
+            return None
+
+    def modify_file_interface_details(self, file_interface_id,
+                                      modify_params):
+        """Perform modify operations on a file interface"""
+
+        try:
+            if not self.module.check_mode:
+                self.file_interface.modify_file_interface(
+                    file_interface_id=file_interface_id,
+                    modify_parameters=modify_params)
+            return True
+        except Exception as e:
+            msg = (f'Failed to modify the file interface instance '
+                   f'with error {str(e)}')
+            LOG.error(msg)
+            self.module.fail_json(msg=msg, **utils.failure_codes(e))
+
+
+def get_powerstore_file_interface_parameters():
+    """This method provides the parameters required for the ansible
+    nas server modules on PowerStore"""
+    return dict(
+        nas_server_name=dict(type='str'),
+        nas_server_id=dict(type='str'),
+        file_interface_id=dict(type='str'),
+        ip_address=dict(type='str'),
+        gateway=dict(type='str'),
+        prefix_length=dict(type='int'),
+        vlan_id=dict(type='int'),
+        ip_port_id=dict(type='str'),
+        is_disabled=dict(required=False, type='bool'),
```

Comment:
The default value for the required will be false. Hence, we can remove it _required=False_